### PR TITLE
fix bias size assert

### LINKED ISSUE #5990: nn.Bilinear without bias broken
**System Information**
------

- OS: OS X 10.13.3
- PyTorch version: 0.4.0a0+f6274a4
- How you installed PyTorch (conda, pip, source): source
- Python version: 3.6.4 (default, Jan  6 2018, 11:51:15)
- CUDA/cuDNN version: NA (On CPU)
- GPU models and configuration: NA (On CPU)
- GCC version (if compiling from source): Apple LLVM version 9.0.0 (clang-900.0.39.2)

**Issue Information**
------

I am trying to use `nn.Bilinear` without a bias, but doing so yields an error.

To reproduce the error run
```python
import torch
import torch.nn as nn

m = nn.Bilinear(20, 30, 40, bias=False)
input1 = torch.randn(128, 20)
input2 = torch.randn(128, 30)
output = m(input1, input2)
```

This yields the error
```bash
(virtualenv) Kellys-MacBook-Pro:sprachspiel kdavis$ python test.py 
Traceback (most recent call last):
  File "test.py", line 7, in <module>
    output = m(input1, input2)
  File "/Users/kdavis/Code/mozilla/sprachspiel/virtualenv/lib/python3.6/site-packages/torch/nn/modules/module.py", line 371, in __call__
    result = self.forward(*input, **kwargs)
  File "/Users/kdavis/Code/mozilla/sprachspiel/virtualenv/lib/python3.6/site-packages/torch/nn/modules/linear.py", line 116, in forward
    return F.bilinear(input1, input2, self.weight, self.bias)
  File "/Users/kdavis/Code/mozilla/sprachspiel/virtualenv/lib/python3.6/site-packages/torch/nn/functional.py", line 1004, in bilinear
    return torch._C._VariableFunctions.bilinear(input1, input2, weight, bias)
RuntimeError: dim() called on undefined Tensor
```

However, running 
```python
import torch
import torch.nn as nn

m = nn.Bilinear(20, 30, 40, bias=True)
input1 = torch.randn(128, 20)
input2 = torch.randn(128, 30)
output = m(input1, input2)
```

works fine.

### PR DESCRIPTION
**Pull request by elanmart**
Fixes https://github.com/pytorch/pytorch/issues/5990

### CODE REVIEW

File: aten/src/ATen/native/Linear.cpp [hunk 1]
```diff
@@ -18,9 +18,12 @@ Tensor bilinear(const Tensor& input1, const Tensor& input2, const Tensor& weight
   AT_ASSERT(input2.size(input2.dim() - 1) == weight.size(2),
             "bilinear(): input2 size does not match weight size: got %lld but expected %lld",
             (long long)input2.size(input2.dim() - 1), (long long)weight.size(2));
-  AT_ASSERT(bias.defined() && bias.size(0) == weight.size(0),
```
Comment (apaszke) [inline aten/src/ATen/native/Linear.cpp]: I think someone just wanted to do `!bias.defined() || bias.size(0) == weight.size(0)`. Can you please use that expression to make the error checking code shorter?

Comment (elanmart) [reply aten/src/ATen/native/Linear.cpp]: sure, I just thought it would be nicer to make it more explicit.

**Comment by apaszke** [comment]
Also, can you please add a test for Bilinear without bias? Just to make sure we won't have a regression again

**Review by ezyang — DISMISSED** [review]
See @apaszke

**Comment by elanmart** [comment]
@ezyang @apaszke Please excuse me the delay. 

How should I read the onnx CI failure?

**Comment by soumith** [comment]
@elanmart the onnx failure means it's asking you to rebase :)

**Comment by elanmart** [comment]
The onnx failure seems unrelated [1], but I can't force the tests to run again :(

[1]
```
E: Failed to fetch http://dl.bintray.com/apache/cassandra/dists/39x/main/binary-amd64/Packages  502  Bad Gateway [IP: 52.27.175.225 80]
```

**Comment by apaszke** [comment]
@onnxbot retest this please

File: test/test_nn.py [hunk 2]
```diff
@@ -4421,6 +4421,22 @@ def test_bilinear(self):
         _assertGradAndGradgradChecks(self, lambda x1, x2: F.bilinear(x1, x2, module.weight, module.bias),
                                      (input1_1, input2_1))
 
+    def test_bilinear_no_bias(self):
```
[the remark this answers is not in the document]
Comment (elanmart) [reply test/test_nn.py]: The only issue is that Bilinear takes two inputs. Should I add a new_module_test and work around that somehow?

File: test/test_nn.py [hunk 3]
```diff
@@ -4421,6 +4421,22 @@ def test_bilinear(self):
...
+        module.weight.data.copy_(module_no_bias.weight)
+
+        input1 = torch.randn(4, 10)
+        input2 = torch.randn(4, 10)
+
+        output = module(input1, input2)
+        output_no_bias = module_no_bias(input1, input2)
+
+        self.assertEqual(output, output_no_bias)
+        output_no_bias.sum().backward()
```
Comment (apaszke) [inline test/test_nn.py]: Can you compare grads too?

**Review by apaszke — APPROVED** [review]

**Pull request merged**